<h1 align="center">🧹 Employee Data Cleaning and Preprocessing</h1>

<p align="center">
<strong>Oasis Infobyte Data Analytics Internship – Task 3</strong>
</p>

<hr>

<h3>📌 Project Overview</h3>

<p>
This project focuses on cleaning and preparing a messy employee dataset for analysis.
The dataset was examined to identify data quality issues such as missing values, incorrect data types, invalid phone number formatting, and potential outliers.
</p>

<p>
Various data cleaning techniques were applied, including missing value imputation, data type conversion, data format standardization, outlier detection using the IQR method, and final data validation.
</p>

<p>
The final outcome of this project is a clean, structured, and analysis-ready employee dataset.
</p>

<h2>🎯 Project Objectives</h2>

<ul>
  <li>Inspect the employee dataset and understand its structure.</li>
  <li>Identify missing values, duplicate records, and data quality issues.</li>
  <li>Handle missing values using appropriate statistical methods.</li>
  <li>Correct inconsistent data types and formatting issues.</li>
  <li>Detect potential outliers using the IQR method.</li>
  <li>Validate the cleaned dataset to ensure data quality.</li>
  <li>Create a clean and analysis-ready dataset for further use.</li>
</ul>

<h2>📚 Import Required Libraries</h2>

<p>
The following libraries are used for data manipulation, analysis, and visualization throughout this project.
</p>

In [1]:
from google.colab import files

uploaded = files.upload()

Saving Messy_Employee_dataset.csv to Messy_Employee_dataset (1).csv


<h2>📁 Load the Dataset</h2>

<p>
The messy employee dataset is uploaded into Google Colab and loaded into a Pandas DataFrame for further inspection and data cleaning.
</p>

In [2]:
import os

os.listdir()

['.config',
 'Messy_Employee_dataset (1).csv',
 'Messy_Employee_dataset.csv',
 'Cleaned_Employee_Dataset.csv',
 'sample_data']

<h3>Loading the Dataset into Pandas</h3>

<p>
The uploaded dataset is loaded into a Pandas DataFrame to examine its records, columns, and overall structure.
</p>

In [3]:
import pandas as pd
import numpy as np

df = pd.read_csv('Messy_Employee_dataset.csv')

df.head()

,Employee_ID,First_Name,Last_Name,Age,Department_Region,Status,Join_Date,Salary,Email,Phone,Performance_Score,Remote_Work
0,EMP1000,Bob,Davis,25.0,DevOps-California,Active,4/2/2021,59767.65,bob.davis@example.com,-1651623197,Average,True
1,EMP1001,Bob,Brown,NaN,Finance-Texas,Active,7/10/2020,65304.66,bob.brown@example.com,-1898471390,Excellent,True
2,EMP1002,Alice,Jones,NaN,Admin-Nevada,Pending,12/7/2023,88145.90,alice.jones@example.com,-5596363211,Good,True
3,EMP1003,Eva,Davis,25.0,Admin-Nevada,Inactive,11/27/2021,69450.99,eva.davis@example.com,-3476490784,Good,True
4,EMP1004,Frank,Williams,25.0,Cloud Tech-Florida,Active,1/5/2022,109324.61,frank.williams@example.com,-1586734256,Poor,False


<h2>🔍 Initial Dataset Inspection</h2>

<p>
Before performing data cleaning, the dataset is inspected to understand its structure, dimensions, column names, data types, and sample records. This initial inspection helps identify potential data quality issues that need to be addressed.
</p>

In [4]:
print("Dataset Shape:", df.shape)

print("\nColumn Names:")
print(df.columns.tolist())

print("\nDataset Information:")
df.info()

print("\nFirst 5 Rows:")
display(df.head())

Dataset Shape: (1020, 12)

Column Names:
['Employee_ID', 'First_Name', 'Last_Name', 'Age', 'Department_Region', 'Status', 'Join_Date', 'Salary', 'Email', 'Phone', 'Performance_Score', 'Remote_Work']

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Employee_ID        1020 non-null   object 
 1   First_Name         1020 non-null   object 
 2   Last_Name          1020 non-null   object 
 3   Age                809 non-null    float64
 4   Department_Region  1020 non-null   object 
 5   Status             1020 non-null   object 
 6   Join_Date          1020 non-null   object 
 7   Salary             996 non-null    float64
 8   Email              1020 non-null   object 
 9   Phone              1020 non-null   int64  
 10  Performance_Score  1020 non-null   object 
 11  Remote_Work        1020 non-null   bool   


,Employee_ID,First_Name,Last_Name,Age,Department_Region,Status,Join_Date,Salary,Email,Phone,Performance_Score,Remote_Work
0,EMP1000,Bob,Davis,25.0,DevOps-California,Active,4/2/2021,59767.65,bob.davis@example.com,-1651623197,Average,True
1,EMP1001,Bob,Brown,NaN,Finance-Texas,Active,7/10/2020,65304.66,bob.brown@example.com,-1898471390,Excellent,True
2,EMP1002,Alice,Jones,NaN,Admin-Nevada,Pending,12/7/2023,88145.90,alice.jones@example.com,-5596363211,Good,True
3,EMP1003,Eva,Davis,25.0,Admin-Nevada,Inactive,11/27/2021,69450.99,eva.davis@example.com,-3476490784,Good,True
4,EMP1004,Frank,Williams,25.0,Cloud Tech-Florida,Active,1/5/2022,109324.61,frank.williams@example.com,-1586734256,Poor,False


<h2>📋 Data Quality Assessment</h2>

<p>
A comprehensive data quality assessment was performed to identify potential issues in the dataset before applying any cleaning techniques.
</p>

<p>
The assessment focuses on:</p>

<ul>
  <li>Missing values</li>
  <li>Missing value percentages</li>
  <li>Duplicate records</li>
  <li>Data types</li>
  <li>Unique values in each column</li>
</ul>

<p>
This analysis helps identify the specific data quality issues that need to be addressed during the cleaning process.
</p>

In [5]:
# ==========================================
# DATA QUALITY REPORT
# ==========================================

print("DATA QUALITY REPORT")
print("=" * 50)

# 1. Missing Values
print("\n1. Missing Values:")
print(df.isnull().sum())

# 2. Missing Value Percentage
print("\n2. Missing Value Percentage:")
print((df.isnull().sum() / len(df) * 100).round(2))

# 3. Duplicate Rows
print("\n3. Number of Duplicate Rows:")
print(df.duplicated().sum())

# 4. Data Types
print("\n4. Data Types:")
print(df.dtypes)

# 5. Unique Values
print("\n5. Unique Values Per Column:")
print(df.nunique())

DATA QUALITY REPORT

1. Missing Values:
Employee_ID            0
First_Name             0
Last_Name              0
Age                  211
Department_Region      0
Status                 0
Join_Date              0
Salary                24
Email                  0
Phone                  0
Performance_Score      0
Remote_Work            0
dtype: int64

2. Missing Value Percentage:
Employee_ID           0.00
First_Name            0.00
Last_Name             0.00
Age                  20.69
Department_Region     0.00
Status                0.00
Join_Date             0.00
Salary                2.35
Email                 0.00
Phone                 0.00
Performance_Score     0.00
Remote_Work           0.00
dtype: float64

3. Number of Duplicate Rows:
0

4. Data Types:
Employee_ID           object
First_Name            object
Last_Name             object
Age                  float64
Department_Region     object
Status                object
Join_Date             object
Salary               float6

<h2>🔎 Detailed Data Exploration</h2>

<p>
After the initial data quality assessment, the dataset was explored in more detail to identify inconsistencies, unusual values, and potential formatting issues within individual columns.
</p>

<p>
The following aspects were examined:</p>

<ul>
  <li>Distribution of categorical variables</li>
  <li>Age value distribution and summary statistics</li>
  <li>Salary distribution and summary statistics</li>
  <li>Department and regional information</li>
  <li>Employee status and performance scores</li>
  <li>Remote work distribution</li>
  <li>Phone number formatting issues</li>
  <li>Sample values from the Join Date column</li>
</ul>

<p>
This detailed exploration helps identify the appropriate cleaning strategy for each data quality issue.
</p>

In [6]:
# ==========================================
# INSPECT CATEGORICAL VALUES
# ==========================================

categorical_columns = [
    'First_Name',
    'Last_Name',
    'Department_Region',
    'Status',
    'Performance_Score',
    'Remote_Work'
]

for column in categorical_columns:
    print(f"\n{'=' * 50}")
    print(f"Column: {column}")
    print(df[column].value_counts(dropna=False))


Column: First_Name
First_Name
Frank      142
Grace      140
Eva        136
Bob        133
Charlie    125
Alice      117
David      116
Heidi      111
Name: count, dtype: int64

Column: Last_Name
Last_Name
Brown       148
Garcia      136
Smith       136
Jones       131
Davis       120
Johnson     118
Miller      116
Williams    115
Name: count, dtype: int64

Column: Department_Region
Department_Region
HR-Florida               41
DevOps-California        35
Sales-Nevada             35
Admin-Nevada             34
Admin-California         34
DevOps-Florida           34
Finance-Illinois         33
DevOps-New York          33
Sales-Florida            33
HR-New York              33
DevOps-Illinois          33
Finance-California       32
Sales-California         31
Sales-Illinois           30
Admin-Illinois           29
Finance-Texas            29
Cloud Tech-Texas         29
Cloud Tech-California    29
Finance-Nevada           29
Cloud Tech-Florida       28
DevOps-Nevada            27
DevOps-

In [7]:
# ==========================================
# INSPECT NUMERICAL COLUMNS
# ==========================================

print("AGE VALUE COUNTS:")
print(df['Age'].value_counts(dropna=False).sort_index())

print("\nSALARY SUMMARY:")
print(df['Salary'].describe())

print("\nPHONE SAMPLE:")
print(df['Phone'].head(20).tolist())

print("\nPHONE MINIMUM:", df['Phone'].min())
print("PHONE MAXIMUM:", df['Phone'].max())

AGE VALUE COUNTS:
Age
25.0    206
30.0    205
35.0    188
40.0    210
NaN     211
Name: count, dtype: int64

SALARY SUMMARY:
count       996.000000
mean      85155.056396
std       19873.727918
min       50047.320000
25%       68392.487500
50%       85547.870000
75%      100974.027500
max      119971.650000
Name: Salary, dtype: float64

PHONE SAMPLE:
[-1651623197, -1898471390, -5596363211, -3476490784, -1586734256, -5409003485, -4518376063, -4134327559, -4177656123, -8156985699, -2720634590, -1238296576, -2429543164, -2669300892, -1734206968, -4215291085, -7713334475, -2639343625, -2044397358, -7107256365]

PHONE MINIMUM: -9994973202
PHONE MAXIMUM: -3896086


In [8]:
# ==========================================
# INSPECT JOIN DATE
# ==========================================

print("Sample Join_Date values:")
print(df['Join_Date'].head(20).tolist())

Sample Join_Date values:
['4/2/2021', '7/10/2020', '12/7/2023', '11/27/2021', '1/5/2022', '6/10/2020', '4/3/2020', '7/17/2022', '12/8/2023', '8/4/2022', '7/6/2024', '9/4/2023', '3/5/2024', '10/10/2020', '6/10/2024', '5/29/2021', '4/23/2024', '11/28/2024', '9/19/2024', '1/2/2020']


<h2>🧩 Missing Value Analysis</h2>

<p>
The initial data quality assessment identified missing values in the <strong>Age</strong> and <strong>Salary</strong> columns. To determine an appropriate strategy for handling these missing values, summary statistics such as the mean and median were examined.
</p>

<ul>
  <li><strong>Age:</strong> 211 missing values</li>
  <li><strong>Salary:</strong> 24 missing values</li>
</ul>

<p>
The median was selected as the imputation method because it provides a robust measure of central tendency and helps reduce the influence of extreme values.
</p>

<h3>📊 Summary Statistics for Missing Value Treatment</h3>

In [9]:
# ==========================================
# MISSING VALUE ANALYSIS
# ==========================================

print("AGE SUMMARY:")
print(df['Age'].describe())

print("\nAge Median:", df['Age'].median())
print("Age Mean:", df['Age'].mean())

print("\nSALARY SUMMARY:")
print(df['Salary'].describe())

print("\nSalary Median:", df['Salary'].median())
print("Salary Mean:", df['Salary'].mean())

AGE SUMMARY:
count    809.000000
mean      32.484549
std        5.656860
min       25.000000
25%       25.000000
50%       30.000000
75%       40.000000
max       40.000000
Name: Age, dtype: float64

Age Median: 30.0
Age Mean: 32.48454882571075

SALARY SUMMARY:
count       996.000000
mean      85155.056396
std       19873.727918
min       50047.320000
25%       68392.487500
50%       85547.870000
75%      100974.027500
max      119971.650000
Name: Salary, dtype: float64

Salary Median: 85547.87
Salary Mean: 85155.05639558233


<h3>🛠️ Handling Missing Values</h3>

<p>
Missing values in the <strong>Age</strong> and <strong>Salary</strong> columns were replaced using their respective median values.
</p>

<p>
After applying median imputation, the dataset was checked again to confirm that no missing values remained.
</p>

In [10]:
# ==========================================
# HANDLE MISSING VALUES
# ==========================================

# Fill missing Age values with the median
df['Age'] = df['Age'].fillna(df['Age'].median())

# Fill missing Salary values with the median
df['Salary'] = df['Salary'].fillna(df['Salary'].median())

# Verify missing values
print("Missing Values After Cleaning:")
print(df.isnull().sum())

Missing Values After Cleaning:
Employee_ID          0
First_Name           0
Last_Name            0
Age                  0
Department_Region    0
Status               0
Join_Date            0
Salary               0
Email                0
Phone                0
Performance_Score    0
Remote_Work          0
dtype: int64


<h2>🔧 Data Type and Format Cleaning</h2>

<p>
After handling missing values, the dataset was further cleaned by correcting data types and standardizing inconsistent formats.
</p>

<h3>📅 Join Date Conversion</h3>

<p>
The <strong>Join_Date</strong> column was initially stored as an object data type. It was converted into a proper datetime format to ensure that date values can be accurately processed and analyzed.
</p>

<h3>📞 Phone Number Formatting</h3>

<p>
The <strong>Phone</strong> column was initially stored as a numeric data type and contained negative values. Since phone numbers are identifiers rather than numerical quantities, the column was converted to text format and negative signs were removed.
</p>

<p>
After applying these transformations, the updated data types and values were verified to ensure successful cleaning.
</p>

In [11]:
# ==========================================
# FIX DATE AND PHONE DATA
# ==========================================

# Convert Join_Date to datetime format
df['Join_Date'] = pd.to_datetime(
    df['Join_Date'],
    errors='coerce'
)

# Convert Phone to string and remove negative sign
df['Phone'] = (
    df['Phone']
    .astype(str)
    .str.replace('-', '', regex=False)
)

# Check results
print("Join_Date Data Type:", df['Join_Date'].dtype)
print("Missing Join_Date Values:", df['Join_Date'].isnull().sum())

print("\nPhone Data Type:", df['Phone'].dtype)
print("\nPhone Sample:")
print(df['Phone'].head())

Join_Date Data Type: datetime64[ns]
Missing Join_Date Values: 0

Phone Data Type: object

Phone Sample:
0    1651623197
1    1898471390
2    5596363211
3    3476490784
4    1586734256
Name: Phone, dtype: object


<h2>📊 Outlier Detection Using the IQR Method</h2>

<p>
Outlier detection was performed on the numerical columns <strong>Age</strong> and <strong>Salary</strong> using the Interquartile Range (IQR) method.
</p>

<p>
The IQR method identifies potential outliers by calculating the first quartile (Q1), third quartile (Q3), and the interquartile range.
</p>

<p>
Values falling below the lower bound or above the upper bound are considered potential outliers.
</p>

<h3>📐 IQR Formula</h3>

<p>
<strong>IQR = Q3 − Q1</strong>
</p>

<p>
<strong>Lower Bound = Q1 − 1.5 × IQR</strong><br>
<strong>Upper Bound = Q3 + 1.5 × IQR</strong>
</p>

In [12]:
# ==========================================
# OUTLIER DETECTION USING IQR
# ==========================================

numerical_columns = ['Age', 'Salary']

for column in numerical_columns:

    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)

    IQR = Q3 - Q1

    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    outliers = df[
        (df[column] < lower_bound) |
        (df[column] > upper_bound)
    ]

    print(f"\n{'=' * 50}")
    print(f"Column: {column}")
    print(f"Q1: {Q1}")
    print(f"Q3: {Q3}")
    print(f"IQR: {IQR}")
    print(f"Lower Bound: {lower_bound}")
    print(f"Upper Bound: {upper_bound}")
    print(f"Number of Outliers: {len(outliers)}")


Column: Age
Q1: 30.0
Q3: 35.0
IQR: 5.0
Lower Bound: 22.5
Upper Bound: 42.5
Number of Outliers: 0

Column: Salary
Q1: 68811.2325
Q3: 100372.6625
IQR: 31561.430000000008
Lower Bound: 21469.087499999987
Upper Bound: 147714.80750000002
Number of Outliers: 0


<h3>🔍 Outlier Detection Results</h3>

<ul>
  <li><strong>Age:</strong> No outliers were detected.</li>
  <li><strong>Salary:</strong> No outliers were detected.</li>
</ul>

<p>
Based on the IQR method, no values required removal or modification. Therefore, all valid records were retained in the dataset.
</p>

<h2>✅ Final Data Validation</h2>

<p>
After completing the data cleaning process, a final validation was performed to ensure that the dataset is clean, consistent, and ready for further analysis.
</p>

<p>
The following checks were performed:</p>

<ul>
  <li>Total remaining missing values</li>
  <li>Duplicate records</li>
  <li>Valid age range</li>
  <li>Valid salary range</li>
  <li>Join date range</li>
  <li>Phone numbers containing negative signs</li>
  <li>Final data types of all columns</li>
</ul>

<p>
This validation confirms that the cleaning steps were successfully applied and that no major data quality issues remain in the dataset.
</p>

In [13]:
# ==========================================
# FINAL DATA VALIDATION
# ==========================================

print("FINAL DATA VALIDATION")
print("=" * 50)

# Missing values
print("\n1. Total Missing Values:")
print(df.isnull().sum().sum())

# Duplicate rows
print("\n2. Duplicate Rows:")
print(df.duplicated().sum())

# Age range
print("\n3. Age Range:")
print("Minimum Age:", df['Age'].min())
print("Maximum Age:", df['Age'].max())

# Salary range
print("\n4. Salary Range:")
print("Minimum Salary:", df['Salary'].min())
print("Maximum Salary:", df['Salary'].max())

# Date range
print("\n5. Join Date Range:")
print("Earliest Date:", df['Join_Date'].min())
print("Latest Date:", df['Join_Date'].max())

# Check phone numbers for negative signs
print("\n6. Phone Numbers with Negative Signs:")
print(df['Phone'].str.startswith('-').sum())

# Data types
print("\n7. Final Data Types:")
print(df.dtypes)

FINAL DATA VALIDATION

1. Total Missing Values:
0

2. Duplicate Rows:
0

3. Age Range:
Minimum Age: 25.0
Maximum Age: 40.0

4. Salary Range:
Minimum Salary: 50047.32
Maximum Salary: 119971.65

5. Join Date Range:
Earliest Date: 2020-01-01 00:00:00
Latest Date: 2024-12-29 00:00:00

6. Phone Numbers with Negative Signs:
0

7. Final Data Types:
Employee_ID                  object
First_Name                   object
Last_Name                    object
Age                         float64
Department_Region            object
Status                       object
Join_Date            datetime64[ns]
Salary                      float64
Email                        object
Phone                        object
Performance_Score            object
Remote_Work                    bool
dtype: object


<h3>📌 Final Validation Results</h3>

<ul>
  <li><strong>Total Missing Values:</strong> 0</li>
  <li><strong>Duplicate Rows:</strong> 0</li>
  <li><strong>Age Range:</strong> 25–40</li>
  <li><strong>Salary Range:</strong> 50,047.32–119,971.65</li>
  <li><strong>Join Date Range:</strong> 2020-01-01 to 2024-12-29</li>
  <li><strong>Phone Numbers with Negative Signs:</strong> 0</li>
  <li><strong>Outliers Detected:</strong> None in Age and Salary</li>
</ul>

<p>
The final validation confirms that the dataset has been successfully cleaned and is now structured, consistent, and ready for further analysis.
</p>

<h2>📊 Before vs After Cleaning Summary</h2>

<p>
To clearly demonstrate the impact of the data cleaning process, a comparison was created between the original dataset and the cleaned dataset.
</p>

<p>
The comparison highlights the major data quality issues identified before cleaning and shows the improvements achieved after applying the cleaning techniques.
</p>

<p>
The following aspects are compared:</p>

<ul>
  <li>Total number of rows and columns</li>
  <li>Missing values</li>
  <li>Duplicate records</li>
  <li>Negative phone number formatting</li>
  <li>Age outliers</li>
  <li>Salary outliers</li>
  <li>Join Date data type</li>
</ul>

In [14]:
# ==========================================
# BEFORE VS AFTER CLEANING SUMMARY
# ==========================================

before_after_summary = pd.DataFrame({
    'Metric': [
        'Total Rows',
        'Total Columns',
        'Missing Values',
        'Duplicate Rows',
        'Negative Phone Numbers',
        'Age Outliers',
        'Salary Outliers',
        'Join_Date Data Type'
    ],

    'Before Cleaning': [
        1020,
        12,
        235,
        0,
        1020,
        0,
        0,
        'object'
    ],

    'After Cleaning': [
        len(df),
        df.shape[1],
        df.isnull().sum().sum(),
        df.duplicated().sum(),
        df['Phone'].str.startswith('-').sum(),
        0,
        0,
        str(df['Join_Date'].dtype)
    ]
})

display(before_after_summary)

,Metric,Before Cleaning,After Cleaning
0,Total Rows,1020,1020
1,Total Columns,12,12
2,Missing Values,235,0
3,Duplicate Rows,0,0
4,Negative Phone Numbers,1020,0
5,Age Outliers,0,0
6,Salary Outliers,0,0
7,Join_Date Data Type,object,datetime64[ns]


<h3>📌 Cleaning Impact Summary</h3>

<ul>
  <li><strong>Missing Values:</strong> Reduced from 235 to 0.</li>
  <li><strong>Duplicate Rows:</strong> No duplicate records were found in the original dataset.</li>
  <li><strong>Phone Number Formatting:</strong> Negative signs were removed from all 1,020 phone numbers, and the column was converted to text format.</li>
  <li><strong>Outliers:</strong> No outliers were detected in the Age or Salary columns using the IQR method.</li>
  <li><strong>Join_Date Data Type:</strong> Successfully converted from <code>object</code> to <code>datetime64[ns]</code>.</li>
</ul>

<p>
Overall, the cleaning process successfully improved the completeness, consistency, and usability of the dataset while retaining all 1,020 original records.
</p>

<h2>💾 Export Cleaned Dataset</h2>

<p>
After completing all data cleaning and validation steps, the final cleaned dataset was exported as a CSV file for future analysis and use.
</p>

<p>
The cleaned dataset retains all <strong>1,020 records</strong> and contains <strong>12 columns</strong>, with missing values handled, data formats corrected, and quality checks completed.
</p>

In [15]:
# ==========================================
# EXPORT CLEANED DATASET
# ==========================================

df.to_csv('Cleaned_Employee_Dataset.csv', index=False)

print("Cleaned dataset exported successfully!")
print("Final Dataset Shape:", df.shape)

Cleaned dataset exported successfully!
Final Dataset Shape: (1020, 12)


<h3>📁 Export Result</h3>

<p>
The cleaned dataset was successfully exported as:</p>

<p>
<strong>Cleaned_Employee_Dataset.csv</strong>
</p>

<p>
This file contains the final cleaned and validated employee data and can be used for further analysis, visualization, or machine learning applications.
</p>

In [16]:
from google.colab import files

files.download('Cleaned_Employee_Dataset.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<h2>💡 Key Findings and Insights</h2>

<ul>
  <li>The original dataset contained <strong>1,020 employee records</strong> and <strong>12 columns</strong>.</li>

  <li>A total of <strong>235 missing values</strong> were identified, including <strong>211 missing values in Age</strong> and <strong>24 missing values in Salary</strong>.</li>

  <li>All missing values were successfully handled using <strong>median imputation</strong>, resulting in <strong>0 remaining missing values</strong>.</li>

  <li>No duplicate records were found in the dataset, so no records needed to be removed.</li>

  <li>The <strong>Join_Date</strong> column was successfully converted from an <strong>object</strong> data type to <strong>datetime64[ns]</strong>.</li>

  <li>All <strong>1,020 phone numbers</strong> contained negative signs. The Phone column was converted to text format, and the negative signs were removed.</li>

  <li>Outlier detection using the <strong>IQR method</strong> found no outliers in either the <strong>Age</strong> or <strong>Salary</strong> columns.</li>

  <li>The final dataset retained all <strong>1,020 original records</strong> and was successfully transformed into a clean and analysis-ready dataset.</li>
</ul>

<h2>🏁 Conclusion</h2>

<p>
This project successfully transformed a messy employee dataset into a clean, structured, and analysis-ready dataset through a systematic data cleaning and preprocessing process.
</p>

<p>
The analysis identified and addressed important data quality issues, including missing values, incorrect data types, and invalid phone number formatting. Missing values in the <strong>Age</strong> and <strong>Salary</strong> columns were handled using median imputation, while the <strong>Join_Date</strong> column was converted into a proper datetime format. Additionally, negative signs were removed from phone numbers after converting the column to an appropriate text format.
</p>

<p>
The dataset was also examined for duplicate records and potential outliers using the IQR method. No duplicate records or outliers were detected, so all <strong>1,020 original records</strong> were retained.
</p>

<p>
Final validation confirmed that the cleaned dataset contains <strong>0 missing values</strong>, corrected data formats, and no remaining negative phone number formatting issues. The final dataset was successfully exported as <strong>Cleaned_Employee_Dataset.csv</strong>.
</p>

<p>
Overall, this project demonstrates the importance of data cleaning and preprocessing in improving data quality and preparing raw data for further analysis, visualization, and machine learning applications.
</p>